confluent_kafka/error.py

```diff
@@ -37,23 +33,49 @@ class ConsumeError(KafkaException):
 
     """
     def __init__(self, error_code, exception=None, message=None):
+        kafka_error = self.constant_to_kafka_error(error_code, exception)
+        self.exception = exception
+
+        super(_KafkaClientError, self).__init__(kafka_error)
+        self.message = str(kafka_error) if message is None else message
+
+    @staticmethod
+    def constant_to_kafka_error(error_code, exception=None):
+        if isinstance(error_code, KafkaError):
+            return error_code
+
         if exception is not None:
-            kafka_error = KafkaError(error_code, repr(exception))
-            self.exception = exception
+            return KafkaError(error_code, repr(exception))
         else:
-            kafka_error = KafkaError(error_code)
-            self.exception = None
-
-        super(ConsumeError, self).__init__(kafka_error)
-        self.message = message
+            return KafkaError(error_code)
 
     @property
     def code(self):
-        return self.code()
+        return self.args[0].code()
 
     @property
     def name(self):
-        return self.name()
+        return self.args[0].name()
+
+
+class ConsumeError(_KafkaClientError):
+    """
+    Wraps all errors encountered during the consumption of a message.
+
+    Note:
+        In the event of a serialization error the original message contents
+        may be retrieved from the ``message`` attribute.
+
+    Args:
+        error_code (KafkaError): Error code indicating the type of error.
```

Comment:
dito

Reply:
Unfortunately we can't change the name as this will break the API.

Reply:
This is part of the experimental serdes feature introduced in v1.4.0, breakage is okay. Let's aim for GA in v1.5.0